client/compare.go

```diff
@@ -0,0 +1,114 @@
+// Copyright 2023 The xline Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package client
+
+import xlineapi "github.com/xline-kv/go-xline/api/xline"
+
+type CompareTarget int
+type CompareResult int
+
+const (
+	CompareVersion CompareTarget = iota
+	CompareCreated
+	CompareModified
+	CompareValue
+)
+
+type Cmp xlineapi.Compare
+
+// nolint: govet
+func Compare(cmp Cmp, result string, v interface{}) Cmp {
```

Comment:
Shall we consider some other relationships, like ">=" and "<="?

Reply:
This should not be supported at this time. Because each symbol is one-to-one with the type definition in the proto file.